cli/tools/publish/mod.rs

```diff
@@ -1166,19 +1164,6 @@ fn collect_excluded_module_diagnostics(
         }
       }
     }
-
-    if !found_outside_specifier {
-      // just in case we didn't find an outside specifier above, add
-      // diagnostics for all the specifiers found outside the package
-      for (specifier, to_package) in outside_specifiers {
-        diagnostics_collector.push(PublishDiagnostic::RelativePackageImport {
-          specifier: specifier.clone(),
-          from_package_name: current_package_name.to_string(),
-          to_package_name: to_package.name.clone(),
-          maybe_referrer: None,
-        });
-      }
-    }
```

Comment:
Walking the graph like above is good enough. I added this just in case, but it wasn't good.